minor improvements for porting guide (#200)
Quansight-Labs/free-threaded-compatibility: docs/porting-extensions.md

code inside a native extension. The lock icon indicates whether the thread holds
the GIL - only one thread can acquire the GIL at a time, indicated by the
fastened lock on the thread calling into the CPython C API. The bottom row of
symbols indicates what work each thread is doing. You can see that even with
the GIL it is possible to get multithreaded parallelism, so long as there are
threads that do not have the GIL acquired and are not waiting to acquire the
GIL. Usually, this means a thread is doing I/O or a long-running calculation
that does not need any state or functionality from the CPython runtime.

In addition to the lock icon indicating whether the GIL is acquired, each thread
icon is either plugged in or unplugged from the interpreter runtime, indicating

> it has an attached or detached thread state. In the GIL-enabled build, an
> attached thread holds the GIL and a detached thread cannot hold the GIL.

Reviewer: This might be a bit clearer?

```suggestion
it has an attached or detached thread state. In the GIL-enabled build, only
one thread can have an attached thread state and hold the GIL, while all
other unplugged threads wait for the GIL to be released.
```

> In the free-threaded build there is no GIL but threads can still either have

Reviewer: ```suggestion
In the free-threaded build the GIL is disabled but threads can still either have
```

attached or detached thread states. As in the GIL-enabled build, only attached
threads can use interpreter state but, because there is no GIL, many threads can
simultaneously call into the CPython C API.

The state of a running free-threaded application is illustrated in the diagram below.

![free-threaded execution diagram](assets/images/free_threaded_diagram.png){ width="600" }
/// caption
A diagramatic snapshot of the state of a multithreaded Python
application running on the free-threaded-enabled interpreter
///

> In the free-threaded build, there is no GIL, so this diagram doesn't have lock

Reviewer: I'm still not sure whether we should be actively making that distinction. If so, we should probably change more places.

```suggestion
In the free-threaded build, the GIL is disabled, so this diagram doesn't have lock
```
